src/collapse/perf.rs

```diff
@@ -202,213 +221,56 @@ impl Collapse for Folder {
         None
     }
 
-    #[cfg(test)]
+    fn is_end_of_stack(&self, line: &[u8]) -> bool {
+        line.iter().all(|b| (*b as char).is_whitespace())
+    }
+
+    fn nstacks_per_job(&self) -> usize {
+        self.nstacks_per_job
+    }
+
     fn set_nstacks_per_job(&mut self, n: usize) {
         self.nstacks_per_job = n;
     }
 
-    #[doc(hidden)]
-    fn set_nthreads(&mut self, n: usize) {
+    fn nthreads(&self) -> usize {
+        self.opt.nthreads
+    }
+
+    fn set_nthreads_(&mut self, n: usize) {
         self.opt.nthreads = n;
-        self.occurrences = Occurrences::new(n);
     }
 }
 
 impl Folder {
-    fn collapse_single_threaded<R>(&mut self, mut reader: R) -> io::Result<()>
+    /// Processes a stack. On success, returns `true` if at end of data; `false` otherwise.
+    fn process_stack<R>(
```

Comment:
I wonder if this should be renamed to `process_single_stack` -- the `return` threw me for a loop!